common/src/main/java/org/opensearch/sql/common/utils/QueryContext.java

```diff
@@ -38,8 +44,7 @@ public static void addRequestId() {
    * @return the current request id from {@link ThreadContext}.
    */
   public static String getRequestId() {
-    final String requestId = ThreadContext.get(REQUEST_ID_KEY);
-    return requestId;
+    return Optional.ofNullable(ThreadContext.get(REQUEST_ID_KEY)).orElseGet(() -> EMPTY_ID);
```

Comment:
This should never happen, right?  
Do we need all this code to protect ourselves from a situation that should never happen, or should we just allow the NullPointerException to get thrown...?

Reply:
Or better yet, let's call `ThreadContext.addRequestId()` if the REQUEST_ID_KEY doesn't exist instead of pass back a 'dummy' ID.

Reply:
This happens in test only. [The test](https://github.com/Bit-Quill/opensearch-project-sql/blob/dev-query-context/legacy/src/test/java/org/opensearch/sql/legacy/unittest/utils/QueryContextTest.java#L48-L51)

Reply:
Don't write production-code for a test.

Reply:
It was written a time ago, I just renamed the file.
Please, see my fix in acf4ffa2.